llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -179,7 +179,9 @@ bool VectorCombine::vectorizeLoadInsert(Instruction &I) {
   // Match insert into fixed vector of scalar value.
   // TODO: Handle non-zero insert index.
   Value *Scalar;
-  if (!match(&I, m_InsertElt(m_Undef(), m_Value(Scalar), m_ZeroInt())) ||
+  UndefValue *BaseVec;
+  if (!match(&I, m_InsertElt(m_UndefValue(BaseVec), m_Value(Scalar),
+                             m_ZeroInt())) ||
```

Comment:
Can we just match poison here? We shouldn't really be getting things like "insert into undef" or "shuffle with undef" anymore nowadays.

Reply:
OK - I'll simplify the fold to just handle m_Poison